src/block-management/block-manager.js

```diff
@@ -366,6 +377,9 @@ export default class BlockManager extends React.Component<PropsType, StateType>
 						this.insertBlocksAfter.bind( this )( value.item.clientId, blocks )
 					}
 					mergeBlocks={ this.mergeBlocks }
+					onReplace={ ( blocks ) =>
+						this.onReplace.bind( this )( value.item.clientId, blocks )
```

Comment:
You shouldn't need to bind to `this` here, just `this.onReplace( value.item.clientId, blocks ) should work